hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -910,7 +910,7 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 
 		awsCluster := controlplaneoperator.AWSCluster(controlPlaneNamespace.Name, hcluster.Name)
-		_, err = controllerutil.CreateOrPatch(ctx, r.Client, awsCluster, func() error {
+		_, err = r.CreateOrUpdate(ctx, r.Client, awsCluster, func() error {
```

Comment:
I changed this, because PATCH means that requests will never conflict which is not what we want, especially not if we increase the odds of having an outdated version because we use a cache.

Afterwards the tests started failing and it turns out that is because CreateOrPatch supports the status subresoruce whereas CreateOrUpdate does not, which is presumably the reason this was used in the first place. I've added support for the status subresource (through a hardcoded list) in the CreateOrUpdate provider and now everything works as exepcted :)

Reply:
ok thanks, I created https://github.com/kubernetes-sigs/controller-runtime/issues/1725